File: FleemanWithNET/Utils/RateCalculator.cs
namespace fleeman_with_dot_net.Utils
{
    public static class RateCalculator
    {
        public static double CalculateRate(DateTime pickup, DateTime dropoff, double dailyRate, double weeklyRate, double monthlyRate)
        {
            var totalDays = (dropoff.Date - pickup.Date).Days + 1;
            double totalCost = 0;

            if (totalDays >= 30)
            {
                int months = totalDays / 30;
                totalCost += months * monthlyRate;
                totalDays %= 30;
            }
            if (totalDays >= 7)
            {
                int weeks = totalDays / 7;
                totalCost += weeks * weeklyRate;
                totalDays %= 7;
            }
            totalCost += totalDays * dailyRate;

            return totalCost;
        }

        public static double CalculateDailyRate(DateTime pickup, DateTime dropoff, double dailyRate)
        {
            var totalDays = (dropoff.Date - pickup.Date).Days + 1;
            return totalDays * dailyRate;
        }

        public static double CalculateAddonRate(DateTime startDate, DateTime endDate, double addonPrice)
        {
            var totalDays = (endDate.Date - startDate.Date).Days + 1;
            return totalDays * addonPrice;
        }

        public static int GetTotalDays(DateTime startDate, DateTime endDate)
        {
            return (endDate.Date - startDate.Date).Days + 1;
        }
    }
}

File: FleemanWithJava/src/main/java/com/example/utils/RateCalculator.java
package com.example.utils;

import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

public class RateCalculator {

    public static float calculateRate(LocalDate pickupDate, LocalDate returnDate,
                                      float dailyRate, float weeklyRate, float monthlyRate) {
        long totalDays = ChronoUnit.DAYS.between(pickupDate, returnDate);
        float totalRate = 0;

        if (totalDays < 7) {
            totalRate = totalDays * dailyRate;
        } else if (totalDays < 30) {
            long weeks = totalDays / 7;
            long remainingDays = totalDays % 7;
            totalRate = (weeks * weeklyRate) + (remainingDays * dailyRate);
        } else {
            long months = totalDays / 30;
            long remainingDays = totalDays % 30;
            long weeks = remainingDays / 7;
            long days = remainingDays % 7;
            totalRate = (months * monthlyRate) + (weeks * weeklyRate) + (days * dailyRate);
        }

        return totalRate;
    }
}

